src/main/kotlin/com/github/mitchellolsthoorn/testgenie/settings/TestGenieSettingsConfigurable.kt

```diff
@@ -23,27 +24,78 @@ class TestGenieSettingsConfigurable : Configurable {
      */
     override fun reset() {
         val settingsState: TestGenieSettingsState = TestGenieSettingsService.getInstance().state!!
-        settingsComponent!!.globalTimeout = settingsState.globalTimeout
         settingsComponent!!.showCoverage = settingsState.showCoverage
+        settingsComponent!!.sandbox = settingsState.sandbox
+        settingsComponent!!.assertions = settingsState.assertions
+        settingsComponent!!.seed = settingsState.seed
+        settingsComponent!!.algorithm = settingsState.algorithm
+        settingsComponent!!.configurationId = settingsState.configurationId
+        settingsComponent!!.clientOnThread = settingsState.clientOnThread
+        settingsComponent!!.junitCheck = settingsState.junitCheck
+        settingsComponent!!.criterionLine = settingsState.criterionLine
+        settingsComponent!!.criterionBranch = settingsState.criterionBranch
+        settingsComponent!!.criterionException = settingsState.criterionException
+        settingsComponent!!.criterionWeakMutation = settingsState.criterionWeakMutation
+        settingsComponent!!.criterionOutput = settingsState.criterionOutput
+        settingsComponent!!.criterionMethod = settingsState.criterionMethod
+        settingsComponent!!.criterionMethodNoException = settingsState.criterionMethodNoException
+        settingsComponent!!.criterionCBranch = settingsState.criterionCBranch
+        settingsComponent!!.minimize = settingsState.minimize
     }
 
     /**
      * Checks if the values of the entries in the settings state are different from the persisted values of these entries.
      */
     override fun isModified(): Boolean {
         val settingsState: TestGenieSettingsState = TestGenieSettingsService.getInstance().state!!
-        var modified: Boolean = settingsComponent!!.globalTimeout != settingsState.globalTimeout
-        modified = modified or (settingsComponent!!.showCoverage != settingsState.showCoverage)
+        var modified: Boolean = settingsComponent!!.showCoverage != settingsState.showCoverage
+        modified = modified or (settingsComponent!!.sandbox != settingsState.sandbox)
+        modified = modified or (settingsComponent!!.assertions != settingsState.assertions)
+        modified = modified or (settingsComponent!!.seed != settingsState.seed)
```

Comment:
Well, i think that here it would be possible to already return true if something along the way has been modified (so that we don't  always do all these checks if we only modify the first box). But that's not an urgent issue and can be addressed in the future if time allows.

Reply:
We may turn this into a `for`-loop eventually, so then it should be possible. Right now, I think that may introduce more unnecessary lines.